jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppContext.java

```diff
@@ -436,12 +436,12 @@ public void preConfigure() throws Exception
 
         // Configure classloader
         _initialClassLoader = getClassLoader();
-        if (!(_initialClassLoader instanceof WebAppClassLoader))
-            setClassLoader(new WebAppClassLoader(_initialClassLoader, this));
+        ClassLoader loader = configureClassLoader(_initialClassLoader);
+        if (loader != _initialClassLoader)
```

Comment:
Why do we actually need `_initialClassLoader`?

Reply:
Because after a start/stop cycle we need to restore the context to as it was configured (by convention)